pkg/apis/pipeline/v1beta1/task_types.go

```diff
@@ -138,6 +138,9 @@ type TaskRef struct {
 	// API version of the referent
 	// +optional
 	APIVersion string `json:"apiVersion,omitempty"`
+	// Image reference that contains the task referenced.
+	// +optional
+	Image string `json:"image,omitempty"`
```

Comment:
i dont see any validation added for this - we'll want to add some at least simple validation (i think https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1beta1/taskrun_validation.go is the place to) and  tests for it; e.g. we'll need to make sure that only one of spec, ref, oci image are specified

https://github.com/tektoncd/pipeline/blob/9e0e8a1a06166124fdf047febf92ebbd9ff58957/pkg/apis/pipeline/v1beta1/taskrun_validation.go#L45-L48